language: python

python:
    - 2.7

sudo: required

services:
  - docker

warnings_are_errors: false

addons:
  apt:
    packages:
      - unixodbc-dev

notifications:
  email:
    on_success: change
    on_failure: change
    
env:
  global:
    - EXAODBC=EXASOL_ODBC-6.0.2
    - EXAODBC_URL=https://www.exasol.com/support/secure/attachment/52896/
    - ODBCINI=$HOME/odbc.ini
    - ODBCSYSINI=$HOME
    - EXAHOST="127.0.0.1:8899"
    - EXAUSER="sys"
    - EXAPW="exasol"
    - ODBC_HOST="127.0.0.1:8899"
    #https://www.exasol.com/support/secure/attachment/52896/EXASOL_ODBC-6.0.2.tar.gz

before_install:
    # test docker
    - docker run --detach --name exasol --privileged -p 127.0.0.1:8899:8888 --stop-timeout 120 exasol/docker-db:latest
    - sleep 120
    - docker exec -it exasol dwad_client list
    - docker inspect exasol

    # move odbcinst.ini to home
    - yes | cp -rf tests/odbcinst.ini $HOME
    - yes | cp -rf tests/odbc.ini $HOME
    
    # get unixODBC
    - wget www.unixodbc.org/unixODBC-2.3.4.tar.gz
    
    # get EXAODBC    
    - wget $EXAODBC_URL$EXAODBC".tar.gz"
   
install:
    - pip install pyodbc
    - pip install pandas
    
    # install unixodbc
    - tar -xzf unixODBC-2.3.4.tar.gz
    - ./unixODBC-2.3.4/configure --prefix=$HOME/unixodbc
    - make
    - make install
    
    # install EXAODBC
    - tar -xzf $EXAODBC".tar.gz"
    - ./$EXAODBC/config_odbc --mode=config --host=$EXAHOST --user=$EXAUSER --password=$EXAPW
    #- ./$EXAODBC/config_odbc --mode=config --host="127.0.0.1:8899" --user=sys --password=exasol
    
    # install py sdk
    - python setup.py install --prefix=$HOME/exa_py

script:
    - PYTHONPATH=$HOME/exa_py/lib/python2.7/site-packages python tests/selftest.py
    - PYTHONPATH=$HOME/exa_py/lib/python2.7/site-packages python tests/data_exchange.py
    
    # udf sdk test and pylint deactivated due to privilege of trail account and toolbox
    #- PYTHONPATH=$HOME/exa_py/lib/python2.7/site-packages python tests/udf_sdk.py
    #- ./tests/pylint.sh
